src/python/pants/backend/jvm/subsystems/scala_coverage_platform.py

```diff
@@ -0,0 +1,124 @@
+# Copyright 2015 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+import os
+from pants.build_graph.injectables_mixin import InjectablesMixin
+from pants.subsystem.subsystem import Subsystem
+from pants.java.jar.jar_dependency import JarDependency
+from pants.backend.jvm.targets.jar_library import JarLibrary
+from pants.build_graph.address import Address
+from typing import Any, Dict, List, Optional, Union
+
+SCOVERAGE = "scoverage"
+blacklist_file = 'new_blacklist_scoverage'
+
+
+class ScalaCoveragePlatform(InjectablesMixin, Subsystem):
+  """The scala coverage platform."""
+
+  options_scope = 'scala-coverage'
+
+  @classmethod
+  def register_options(cls, register):
+    super().register_options(register)
+    register('--enable-scoverage',
+      default=False,
+      type=bool,
+      help='Specifies whether to generate scoverage reports for scala test targets.'
```

Comment:
This should be consumed from this subsystem instead: https://github.com/pantsbuild/pants/blob/4435b8ea14cc753d21f0c22377140cab2156c593/src/python/pants/backend/jvm/tasks/coverage/manager.py#L73-L74

Reply:
If I understand correctly, `manager.py` is initialized by `junit_run.py` at runtime only. However, I also need this option (`enable-scoverage`) at compile time to determine whether to instrument the targets or not. That is why I kept it here. Am I thinking about this right?

Reply:
I think I see what is happening here: Even though the CodeCoverage subsystem gets initialized, none of its options are registered. This is because the subsystem is using `def register_junit_options(register, register_jvm_tool):` instead of `def register_options(cls, register):`

while the options under the latter (`register_options`) gets automatically registered, the former (`register_junit_options`) requires to be called by some other code, e.g, this is how it is called in `junit_run.py`: https://github.com/pantsbuild/pants/blob/35be5116da7c98f6d789ebff9baf58a8379afdf5/src/python/pants/backend/jvm/tasks/junit_run.py#L99-L102

And when it is called by some other class,( `junit` in this example), the scope of that class gets prepended to the scope of `coverage`, thus registering the options as `--test-junit-coverage`

Consequently, setting `--scoverage-coverage=True` (scope at compile time) does not automatically set `--test-junit-coverage=True`(scope at runtime) (and vice-versa)